dpe/src/commands/derive_child.rs

```diff
@@ -92,13 +89,23 @@ impl<C: Crypto, P: Platform> CommandExecution<C, P> for DeriveChildCmd {
         locality: u32,
     ) -> Result<Response, DpeErrorCode> {
         // Make sure the operation is supported.
-        if !dpe.support.internal_info && self.uses_internal_info_input()
-            || !dpe.support.internal_dice && self.uses_internal_dice_input()
+        if (!dpe.support.internal_info && self.uses_internal_info_input())
+            || (!dpe.support.internal_dice && self.uses_internal_dice_input())
         {
-            return Err(DpeErrorCode::InvalidArgument);
+            return Err(DpeErrorCode::ArgumentNotSupported);
+        }
+
+        if (!dpe.support.is_ca && self.allows_ca()) || (!dpe.support.x509 && self.allows_x509()) {
```

Comment:
```allows_ca``` (and ```allows_x509```) sound like they are inclusive, i.e., the argument can be either a CA or not. This test however, looks like the argument is *required* to be a CA. Is that the intention?